components/script/dom/document.rs

```diff
@@ -575,10 +575,19 @@ impl Document {
 
     // https://html.spec.whatwg.org/multipage/#fallback-base-url
     pub fn fallback_base_url(&self) -> ServoUrl {
-        // Step 1: iframe srcdoc (#4767).
-        // Step 2: about:blank with a creator browsing context.
-        // Step 3.
-        self.url()
+        let document_url = self.url();
+        if let Some(browsing_context) = self.browsing_context() {
+            // Step 1: If document is an iframe srcdoc document, then return the
```

Comment:
We support srcdoc iframes, so we could get `browsing_context.parent().and_then(|parent| parent.document()).map(|doc| doc.base_url())` and implement this missing step.

Reply:
Does having a nested browsing context correctly identify if the document is an iframe srcdoc here?

Reply:
We'll want to check for an `about:srcdoc` url as well, given https://github.com/servo/servo/blob/master/components/script/script_thread.rs#L3923.

Reply:
Right. I have updated the changes accordingly.